### PR TITLE
Update the by-laws to introduce the OTC concept

### PR DESCRIPTION
**Pull request by mattcaswell**
We split the responsibilities of the current OMC into two different
groups - the OMC and the OTC (OpenSSL Technical Committee). The OMC still
retains its overall management function but the OTC becomes responsible
for technical decision making.

PR reviews will then require approval from an OTC member instead of an
OMC member.

### CODE REVIEW

**Comment by mattcaswell** [comment]
Note there will be further updates required to other policies etc if this gets approved. This PR just focuses on the bylaw changes.

**Review by t-j-h — APPROVED** [review]

**Comment by paulidale** [comment]
Although this can be merged now, a week's grace will be available for comments and suggestions before it is merged.

**Comment by mattcaswell** [comment]
> Although this can be merged now, a week's grace will be available for comments and suggestions before it is merged.

Not forgetting that an OMC vote is also required after comments/suggestions.

**Comment by beldmit** [comment]
I'm still not sure whether we can strictly put apart the OMC and OTC spheres of responsibility in the part of features supported. "Features" are a bit too wide word covering both features like CMP and something relatively small (e.g. X509 extensions). The 1st is definitely in the sphere of OMC, the 2nd seems not.

**Comment by mattcaswell** [comment]
> I'm still not sure whether we can strictly put apart the OMC and OTC spheres of responsibility in the part of features supported. "Features" are a bit too wide word covering both features like CMP and something relatively small (e.g. X509 extensions). The 1st is definitely in the sphere of OMC, the 2nd seems not.

Do you have a suggestion as to how we could change the wording?

I notice that the word "feature" only appears once, i.e. in the context of "feature requirements" being an OMC responsibility. In my mind this split seems to work whether the feature is large or small. If small the "requirement" might be as simple as "should we have this feature or not?".

**Comment by beldmit** [comment]
> Do you have a suggestion as to how we could change the wording?

"key features"/"major features"?

**Comment by t-j-h** [comment]
I think we are in a context here where we need to see if there actually is an issue before adjusting from the proposed wording. The OTC is focused on the "how" and the OMC on the "what" and that distinction is reflected here.  If we put in "key" or "major" then we leave a significant ambiguity of responsibility, and in my view keeping this clear distinction makes sense. If it turns into an issue down the track with how the OMC and OTC interact then we can adjust it then. 

i.e. this seems like a premature "optimisation" without any performance measurements :-)

**Comment by levitte** [comment]
> i.e. this seems like a premature "optimisation" without any performance measurements :-)

I can get along with that thinking.

**Comment by beldmit** [comment]
OK. Let's leave it and hope for the best.

**Review by iamamoose — APPROVED** [review]

**Review by levitte — APPROVED** [review]

**Comment by mspncp** [comment]
I know I'm late to the game, but I agree with @beldmit that the vague distinction between technical and non-technical decisions is not optimal. Naturally, **a lot** of the strategical decisions of the OMC will be of **technical** nature.

Instead, it would be good to clarify the responsibilites of the two entities and their relationship a little bit more. Essentially, they can be summarized as follows:

- The OMC makes the decisions about the overall strategy and the direction in which the project is heading.
- The OTC takes care of the daily technical questions and decisions on behalf of the OMC.
- In general, the OMC will not interfere with the OTC. However, if OTC decisions contradict OMC policies or decisions, the OMC is the ultimate authority and can overrule OTC decisons by an OMC vote.

The following patch shows a possible rewording:

```patch
From 2138a7e4a19ea2aa143186312cb6ba0c6bf3b418 Mon Sep 17 00:00:00 2001
From: "Dr. Matthias St. Pierre" <Matthias.St.Pierre@ncp-e.com>
Date: Thu, 28 Nov 2019 19:33:51 +0100
Subject: [PATCH] fixup! Update the by-laws to introduce the OTC concept

---
 policies/omc-bylaws.html | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/policies/omc-bylaws.html b/policies/omc-bylaws.html
index ecc9b88..03d05dc 100644
--- a/policies/omc-bylaws.html
+++ b/policies/omc-bylaws.html
@@ -71,7 +71,7 @@
 
           <p>The OMC:</p>
           <ul>
-            <li>makes all non-technical decisions regarding management and strategic direction
+            <li>makes all decisions regarding management and strategic direction
             of the project; including:
             <ul>
               <li>business requirements;</li>
@@ -111,6 +111,13 @@
           but the ones that count in order to participate in the OMC
           decision-making process are the ones listed below.</p>
 
+          <p>In general, the OMC will leave technical decisions to the so-called
+          OpenSSL Technical Committee (OTC, see below) and not participate in
+          discussions related to development and documention of the OpenSSL
+          Toolkit. In exceptional cases however an OTC vote can be overruled
+          by an OMC vote. Such an exceptional case would be for example if an
+          OTC decision stands contrary to OMC policies or decisions.
+
           <p>OMC members may become inactive. In order to remain active a member
           must, in any calendar quarter, contribute by:</p>
           <ul>
-- 
2.23.0

```


**P.S:** IMHO it would even make sense to demand a two-third OMC vote in order to overrule an OTC vote. This would ensure that this type of vote remains exceptional. Otherwise, OTC votes and OMC votes might become interchangeable ad libitum, depending on which gremium offers higher chances for winning the vote.

**Comment by paulidale** [comment]
How to get this one moving again?
It's got the required approvals, it also has what I consider to be a worthwhile change suggested.

**Comment by kroeckx** [comment]
Approvals here is not enough to merge it, it needs to be voted
on. It's not because some people approved it that you can't make
changes before we vote on it.

As long as who voted what is actually not public information,
I would also prefer that people didn't actually approve things we
need to vote on.

**Comment by paulidale** [comment]
We need an indication as to when to vote, the approvals do that.
We need to call a vote, should @mspncp's changes be incorporated?

**Comment by mattcaswell** [comment]
> We need an indication as to when to vote, the approvals do that.

I agree. I think the approvals are a useful indication as to whether we're headed in the right direction.

> We need to call a vote, should @mspncp's changes be incorporated?

I think they are a useful clarification. I'll incorporate them on Monday unless feedback here in the meantime suggests I shouldn't. I might make some minor editorial tweaks.

**Comment by mspncp** [comment]
Note: Looking back at my proposal, I would omit the ‘so-called’ and shorten the sentence to

> In general, the OMC will leave technical decisions to the OpenSSL Technical Committee (OTC) and ...

**Comment by mspncp** [comment]
> IMHO it would even make sense to demand a two-third OMC vote in order to overrule an OTC vote. This would ensure that this type of vote remains exceptional. Otherwise, OTC votes and OMC votes might become interchangeable ad libitum, depending on which gremium offers higher chances for winning the vote.

Any thoughts/opinions about the two-third rule?

**Comment by t-j-h** [comment]
Again I think you are prematurely optimising things without any data to indicate that there is an issue.

The earlier comment you made of "the OMC is the ultimate authority and can overrule OTC decisions by an OMC vote" is rather clear that you get this distinction. 

There is no intended barrier on the OMC in terms of making decisions. 
The OMC has its own voting process for things and that should be sufficient for making such decisions. 

Keep in mind that the OTC is meant to be entirely technically focused. 

If we find that things aren't workable then we can adjust them - but we first need to get the OTC in place and operating and get experience at how it works compared to how we perceived it will work. 
That experience needs to come on the basis of actually working with what we have outline for a reasonable period of time.

**Comment by mattcaswell** [comment]
> I think they are a useful clarification. I'll incorporate them on Monday unless feedback here in the meantime suggests I shouldn't. I might make some minor editorial tweaks.

I have incorporated the patch. I removed the "so called" bit, tweaked the date at the beginning of the document and added a missing `</p>`

**Comment by mattcaswell** [comment]
The vote to adopt this has been started. I will report back here with the results:

> topic: Change the project bylaws to the text shown here:
>        https://github.com/openssl/web/pull/143 (as of commit 0d86944dca1)
> Proposed by Matt Caswell
> Public: yes
> opened: 2019-12-10

**Comment by mattcaswell** [comment]
This vote has now closed. The results were:

+1: 7 votes
 0: 0 votes
-1: 0 votes

So the vote has passed unanimously.

Since there were updates made between the approvals received in this PR and the version that was finally voted on - please can someone re-approve this so I can push?

**Review by levitte — APPROVED** [review]

**Comment by mattcaswell** [comment]
Pushed! Thanks.

**Pull request closed without merging**
